File: subly_ios/Subly/Subly/Domain/Repositories/AuthRepository.swift
import Foundation

protocol AuthRepository {
    var currentUser: User? { get }
    var isLoggedIn: Bool { get }
    func signInWithEmail(email: String, password: String) async -> Result<User, Error>
    func registerWithEmail(email: String, password: String) async -> Result<User, Error>
    func signInWithGoogle(idToken: String) async -> Result<User, Error>
    func signOut() async
    func syncOnLogin(uid: String) async
}

File: subly_android/app/src/main/java/net/cynreub/subly/domain/repository/AuthRepository.kt
package net.cynreub.subly.domain.repository

import net.cynreub.subly.domain.model.User

interface AuthRepository {
    val currentUser: User?
    val isLoggedIn: Boolean
    suspend fun signInWithEmail(email: String, password: String): Result<User>
    suspend fun registerWithEmail(email: String, password: String): Result<User>
    suspend fun signInWithGoogle(idToken: String): Result<User>
    suspend fun signOut()
    suspend fun syncOnLogin(uid: String)
}
